src/App.js

```diff
@@ -2,26 +2,20 @@
 /** @jsx jsx */
 import { jsx } from 'theme-ui';
 
-import { Box, Flex } from '@theme-ui/components';
+import { Flex } from '@theme-ui/components';
 import PropTypes from 'prop-types';
+import { useState } from 'react';
 import { BrowserRouter as Router, Switch, Redirect, Route } from 'react-router-dom';
 
-import useLocalStorage from './use-local-storage';
-import initial from './default-settings';
-
 import About from './ui/about';
 import OpencastHeader from './ui/opencast-header';
-import Settings from './ui/settings';
+import SettingsPage from './ui/settings/page.js';
 import Studio from './ui/studio';
 
-const SETTINGS_KEY = 'ocStudioSettings';
-
-function App({ defaultSettings = initial }) {
-  const [settings, updateSettings] = useLocalStorage(SETTINGS_KEY, defaultSettings);
 
-  const handleUpdate = data => {
-    updateSettings(prevState => ({ ...prevState, ...data, connected: true }));
-  };
+function App({ settingsManager }) {
+  const [settings, updateSettings] = useState(settingsManager.settings());
```

Comment:
As we discussed: `setState` (or its hook equivalent) needs to be called so that the app redraws. And using the actual settings object as state seems "more correct" than the settings manager, since most components just depend on the settings, not the manager.